app/server.go

```diff
@@ -3,44 +3,52 @@ package app
 import (
 	"context"
 	"fmt"
+	"io"
 	"os"
 	"os/signal"
 	"runtime"
 	"syscall"
 	"time"
 
+	"cloud.google.com/go/pubsub"
 	"github.com/raystack/raccoon/collection"
 	"github.com/raystack/raccoon/config"
 	"github.com/raystack/raccoon/logger"
 	"github.com/raystack/raccoon/metrics"
 	"github.com/raystack/raccoon/publisher"
 	"github.com/raystack/raccoon/services"
 	"github.com/raystack/raccoon/worker"
+	"google.golang.org/api/option"
 )
 
+type Publisher interface {
+	worker.Producer
+	io.Closer
+}
+
 // StartServer starts the server
 func StartServer(ctx context.Context, cancel context.CancelFunc) {
 	bufferChannel := make(chan collection.CollectRequest, config.Worker.ChannelSize)
 	httpServices := services.Create(bufferChannel)
 	logger.Info("Start Server -->")
 	httpServices.Start(ctx, cancel)
-	logger.Info("Start publisher -->")
-	kPublisher, err := publisher.NewKafka()
+	logger.Infof("Start publisher --> %s", config.Publisher)
+	publisher, err := initPublisher()
 	if err != nil {
-		logger.Error("Error creating kafka producer", err)
+		logger.Errorf("Error creating %q publisher: %v\n", config.Publisher, err)
 		logger.Info("Exiting server")
 		os.Exit(0)
 	}
 
 	logger.Info("Start worker -->")
-	workerPool := worker.CreateWorkerPool(config.Worker.WorkersPoolSize, bufferChannel, config.Worker.DeliveryChannelSize, kPublisher)
+	workerPool := worker.CreateWorkerPool(config.Worker.WorkersPoolSize, bufferChannel, config.Worker.DeliveryChannelSize, publisher)
 	workerPool.StartWorkers()
-	go kPublisher.ReportStats()
+
 	go reportProcMetrics()
-	go shutDownServer(ctx, cancel, httpServices, bufferChannel, workerPool, kPublisher)
+	go shutDownServer(ctx, cancel, httpServices, bufferChannel, workerPool, publisher)
 }
 
-func shutDownServer(ctx context.Context, cancel context.CancelFunc, httpServices services.Services, bufferChannel chan collection.CollectRequest, workerPool *worker.Pool, kp *publisher.Kafka) {
+func shutDownServer(ctx context.Context, cancel context.CancelFunc, httpServices services.Services, bufferChannel chan collection.CollectRequest, workerPool *worker.Pool, kp Publisher) {
```

Comment:
@turtleDev Let's update short var as well `kp`

Reply:
renamed to `pub`.